lib/rules/no-restricted-imports.js

```diff
@@ -446,11 +531,32 @@ module.exports = {
 		 * @param {Object} group contains an Ignore instance for paths, the customMessage to show on failure,
 		 * and any restricted import names that have been specified in the config
 		 * @param {Map<string,Object[]>} importNames Map of import names that are being imported
+		 * @param {string} importSource the import source string
 		 * @returns {void}
 		 * @private
 		 */
-		function reportPathForPatterns(node, group, importNames) {
-			const importSource = node.source.value.trim();
+		function reportPathForPatterns(node, group, importNames, importSource) {
+			// Use provided importSource or fallback to node.source.value for backward compatibility
+			const actualImportSource = importSource || node.source.value.trim();
+
+			// Skip if this is a type-only import and it's allowed
+			if (
+				node.type === "ImportDeclaration" &&
+				isTypeOnlyImport(node) &&
+				group.allowTypeImports
+			) {
+				return;
+			}
```

Comment:
I think this would be a false positive:

```ts
/* eslint no-restricted-imports: ["error", {
  "patterns": [{
    "group": ["f*"],
	"allowTypeImports": true
  }]
}]
*/

import type fs = require("fs");

```

Reply:
Fixed in https://github.com/eslint/eslint/pull/19562/commits/59711edf410e6ebc6b4d3d729b6735b890addec1